### PR TITLE
IFPRI Whitelabeling

### PR DESCRIPTION
**Pull request by adamgilg**
This PR adds whitelabeling capabilities across the app. Now every instance of the word `Dojo` has been replaced with a component or logic that swaps it based on the COMPANY_BRANDING environment variable (we can easily change this env variable name as it's only referenced in two UI files). Some other copy has also been changed, as has the landing page and navbar styling. 

For styling, I created two custom themes that we swap out in `theme.js`. Both of these implement a `theme.custom` styling namespace that lets us access these styles from anywhere with access to MUI theme context. 

I created a few very small reusable utilities to make the whitelabeling as simple as possible for a starting point, as well as a `usePageTitle` hook that I used to replace every instance of our `useEffect document.title = title` pattern.

### CODE REVIEW

File: ui/client/components/CSVDownload.js [hunk 1]
```diff
@@ -36,17 +36,17 @@ function CSVDownload({ resource, index = 'indicators' }) {
   const theme = useTheme();
   const { classes } = useStyles();
 
-  const isDesktop = useMediaQuery('(min-width:800px)'); 
+  const isDesktop = useMediaQuery('(min-width:800px)');
```
Comment (adamgilg) [inline ui/client/components/CSVDownload.js:39]: All the changes in this file are auto formatting changes from running eslint.

File: ui/client/admin.js [hunk 2]
```diff
@@ -42,6 +43,7 @@ const Admin = () => {
   const [nodeInfo, setNodeInfo] = useState([]);
   const [shutDownFailed, setShutDownFailed] = useState(false);
   const [alertMessage, setAlertMessage] = useState('');
+  usePageTitle({ title: 'Admin' });
```
Comment (adamgilg) [inline ui/client/admin.js:46]: This is the standard example of the change made to most of the files in this PR: adding this hook, removing the useEffect.

File: ui/client/index.js [hunk 3]
```diff
@@ -78,10 +78,31 @@ export default function Main() {
...
+    link.href = `/assets/${faviconFileName}`;
+  } else {
+    const newLink = document.createElement('link');
+    newLink.rel = 'icon';
+    newLink.href = `/assets/${faviconFileName}`;
+    document.head.appendChild(newLink);
+  }
+};
+
+if (process.env.COMPANY_BRANDING === 'ifpri') {
```
Comment (adamgilg) [inline ui/client/index.js:93]: I played around with changing this to use the functions in Branding.js, but it ended up making it harder to read and didn't reduce much code, so I kept it as is.

File: ui/client/theme.js [hunk 4]
```diff
@@ -111,3 +114,54 @@ export default createTheme(theme, {
...
+    landing: {
+      backgroundColor: '#f5f4f0',
+      color: 'black',
+    },
+    button: {
+      color: 'white',
+      backgroundColor: '#429d46',
+    },
+  },
+  palette: {
```
Comment (adamgilg) [inline ui/client/theme.js:151]: override the mui palette.secondary color here

File: ui/client/theme.js [hunk 5]
```diff
@@ -111,3 +114,54 @@ export default createTheme(theme, {
...
+      dark: '#2e6e31',
+      constrastText: '#fff',
+    }
+  }
+};
+
+function createCustomTheme() {
+  const themeVariant = BrandSwap({ dojo: dojoTheme, ifpri: ifpriTheme });
+
+  return createTheme(deepmerge(baseTheme, themeVariant));
```
Comment (adamgilg) [inline ui/client/theme.js:164]: deepmerge is the MUI utility for merging themes: [see the first example here](https://mui.com/material-ui/customization/theming/#api)

File: ui/webpack.config.js [hunk 6]
```diff
@@ -41,9 +41,13 @@ module.exports = {
                 sideEffects: true
             },
           {
-            test: /\.(png|jpg|gif|woff|woff2|eot|ttf|svg|mp4)$/,
+            test: /\.(png|jpg|gif|woff|woff2|eot|ttf|mp4)$/,
             loader: 'file-loader',
           },
+          {
+            test: /\.svg$/,
+            use: ['@svgr/webpack', 'file-loader'],
```
Comment (adamgilg) [inline ui/webpack.config.js:49]: Using this to load in the IFPRI svgs into <SvgIcon>: https://mui.com/material-ui/icons/#component-prop

File: ui/client/components/uiComponents/Branding.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
```
Comment (adamgilg) [inline ui/client/components/uiComponents/Branding.js:1]: No comment really needed here. Simple utility that's used to support every instance (except the `index.js` favicon switch) where we want to check the `COMPANY_BRANDING` environment variable. Two other small utility functions to translate the env variable into the formatted names, for use both in and outside of JSX.

**Review by ccjoel — COMMENTED** [review]
🚀

**Pull request merged**